pkg/expression/scalar_function.go

```diff
@@ -418,9 +418,8 @@ func (sf *ScalarFunction) Eval(ctx EvalContext, row chunk.Row) (d types.Datum, e
 		str, isNull, err = sf.EvalString(ctx, row)
 		if !isNull && err == nil && tp.GetType() == mysql.TypeEnum {
 			res, err = types.ParseEnum(tp.GetElems(), str, tp.GetCollate())
-			if sc := ctx.GetSessionVars().StmtCtx; sc != nil {
-				err = sc.HandleTruncate(err)
-			}
+			tc := typeCtx(ctx)
```

Comment:
I think the `ctx.GetSessionVars().StmtCtx` will always return a no-nil value because it is assigned in `NewSessionVars` and no else changes its value.

Reply:
lgtm